django/core/handlers/asgi.py

```diff
@@ -184,19 +184,22 @@ async def handle(self, scope, receive, send):
             asyncio.create_task(self.listen_for_disconnect(receive)),
         ]
         done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-        # Cancel the pending task and await canceling
-        for pending_task in pending:
-            pending_task.cancel()
-            try:
-                await pending_task
-            except asyncio.exceptions.CancelledError:
-                pass
-        # Catch RequestAborted raised from possible disconnect
+        done, pending = done.pop(), pending.pop()
+        # Allow views to handle cancellation.
+        pending.cancel()
         try:
-            response = done.pop().result()
-        except (RequestAborted, AssertionError):
+            await pending
+        except asyncio.CancelledError:
+            # Task re-raised the CancelledError as expected.
+            pass
+        try:
+            response = done.result()
+        except RequestAborted:
             body_file.close()
             return
+        except AssertionError:
+            body_file.close()
+            raise
```

Comment:
We **could** handle this with `RequestAborted` above but it would be a programming error in the protocol server (e.g. Daphne/Uvicorn) and I'd just want to be uncaught. (It's a genuine _should **never** happen_, so an assertion is the right option.) 

(We could drop it entirely... equally)

Reply:
Yeah. I actually think It's an awesome catch.